oci/layout/oci_transport.go

```diff
@@ -88,6 +88,12 @@ type ociReference struct {
 	tag         string
 }
 
+// GetOciReference returns the dir, resolvedDir, and tag of ociReference
+// so that it is accessible in oci/archive
+func GetOciReference(t types.ImageReference) (string, string, string) {
+	return t.(ociReference).dir, t.(ociReference).resolvedDir, t.(ociReference).tag
```

Comment:
This would just panic if the caller supplied a wrong type.  But overall, after the use in `archive.ParseReference` is removed, this won’t be needed: when the caller creates a reference for the temporary directory, it can just remember the directory path.  That wastes a tiny bit of memory but the code is easier to follow and clearly can’t panic.